connectors/agent/protocol.py

```diff
@@ -0,0 +1,68 @@
+from elastic_agent_client.generated import elastic_agent_client_pb2 as proto
+from elastic_agent_client.handler.action import BaseActionHandler
+from elastic_agent_client.handler.checkin import BaseCheckinHandler
+from elastic_agent_client.util.logger import logger
+
+
+class ConnectorActionHandler(BaseActionHandler):
+    """Class handling Agent actions.
+
+    As there are no actions that we can respond to, we don't actually do anything here.
+    """
+
+    async def handle_action(self, action: proto.ActionRequest):
+        """Implementation of BaseActionHandler.handle_action
+
+        Right now does nothing as Connectors Service has no actions to respond to.
+        """
+        msg = (
+            f"This connector component can't handle action requests. Received: {action}"
+        )
+        raise NotImplementedError(msg)
+
+
+class ConnectorCheckinHandler(BaseCheckinHandler):
+    """Class handling to Agent check-in events.
+
+    Agent sends check-in events from time to time that might contain
```

Comment:
from time to time -> "at least every 30s"  https://github.com/elastic/elastic-agent-client/blob/main/elastic-agent-client.proto#L29

So if apply_from_client blocks for more than 30s the component is marked as failed